contract/src/msgs/data_requests/query.rs

```diff
@@ -1,8 +1,36 @@
-use super::{msgs::data_requests::query::QueryMsg, *};
+use execute::commit_result::VerifyCommit;
+
+use super::{
+    msgs::data_requests::{execute::commit_result, query::QueryMsg},
+    *,
+};
 
 impl QueryHandler for QueryMsg {
-    fn query(self, deps: Deps, _env: Env) -> Result<Binary, ContractError> {
+    fn query(self, deps: Deps, env: Env) -> Result<Binary, ContractError> {
         let binary = match self {
+            QueryMsg::CanExecutorCommit {
+                dr_id,
+                public_key,
+                commitment,
+                proof,
+            } => {
+                let dr = state::may_load_request(deps.storage, &Hash::from_hex_str(&dr_id)?)?;
+                let valid = dr.map_or(false, |dr| {
+                    let commit_msg = commit_result::Execute {
+                        dr_id,
+                        commitment,
+                        public_key,
+                        proof,
+                    };
+                    commit_msg.verify_commit(deps, &env, &dr).is_ok()
+                });
+                to_json_binary(&valid)?
```

Comment:
This is an `Option<bool>` where as in common the query definition we have that it returns a `bool`.

Reply:
It is already a bool though? 
<img width="232" alt="Screenshot 2025-01-07 at 17 57 52" src="https://github.com/user-attachments/assets/e6b93e73-8648-4014-8153-553548545928" />

Reply:
Oh my bad, thought it would have been an `option` I misread `map_or` as `map` :sweat_smile: